### PR TITLE
[Stage of Stage] Add NodesList

### PR DESCRIPTION
**Pull request by jfurrow**
This introduces the `NodesList` which appears for each stage of the installer and displays the status of each node.

It renders gradient masks when the user has scrolled away from the top or bottom of the list, as well as when animating open or closed.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @rcorral

File: src/js/components/NodesList.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+    let containerHeight = Math.min(
+      this.refs.nodesList.offsetHeight,
+      MAX_HEIGHT
+    );
+
+    // The height of the element must be determined and set explicitly so that
+    // we can animate the height with CSS.
+    if (this.props.visible && this.state.containerHeight === 0) {
+      /* eslint-disable */
```
Comment (kennyt) [inline src/js/components/NodesList.js:51]: why is this needed?

File: src/js/components/NodesList.js [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    if (!this.refs.gemini) {
+      this.setState({
+        shouldRenderBottomMask: false,
+        shouldRenderTopMask: false
+      });
+
+      return;
+    }
+
+    if (!this.state.shouldRenderTopMask && this.refs.gemini.refs['scroll-view'].scrollTop > 0) {
```
Comment (kennyt) [inline src/js/components/NodesList.js]: can we cache `this.state` and `this.refs`

File: src/js/components/NodesList.js (hunk 1 above)
Comment (jfurrow) [inline src/js/components/NodesList.js:51]: We have an eslint rule that disallows `setState` in `componentDidUpdate`

File: src/js/components/NodesList.js (hunk 2 above)
Comment (kennyt) [inline src/js/components/NodesList.js]: let's also cache the `scroll-view` element

File: src/js/components/NodesList.js [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+      this.setState({shouldRenderTopMask: false});
+    } else if (this.state.shouldRenderTopMask
+      && this.refs.gemini.refs['scroll-view'].scrollTop + this.state.containerHeight
+      >= this.refs.nodesList.offsetHeight + LIST_PADDING) {
+      this.setState({shouldRenderBottomMask: false});
+    }
+  }
+
+  getNodesList(nodes) {
+    let sortedNodes = nodes.sort((a, b) => {
```
Comment (kennyt) [inline src/js/components/NodesList.js]: we need the fat arrow?

**Comment by kennyt** [comment]
after the comments, i'll merge

File: src/js/components/NodesList.js (hunk 1 above)
Comment (jfurrow) [inline src/js/components/NodesList.js]: Done!

File: src/js/components/NodesList.js (hunk 2 above)
Comment (jfurrow) [inline src/js/components/NodesList.js]: Done!

File: src/js/components/NodesList.js (hunk 3 above)
Comment (jfurrow) [inline src/js/components/NodesList.js]: Nope, done!
